code/game/objects/items/bodybag.dm

```diff
@@ -89,4 +92,74 @@
 	// Make it possible to escape from bodybags in morgues and crematoriums
 	if(loc && (isturf(loc) || istype(loc, /obj/structure/morgue) || istype(loc, /obj/structure/crematorium)))
 		if(!open())
-			to_chat(user, "<span class='notice'>It won't budge!</span>")
\ No newline at end of file
+			to_chat(user, "<span class='notice'>It won't budge!</span>")
+
+/obj/item/bodybag/cryobag
+	name = "stasis bag"
+	desc = "A folded, non-reusable bag designed for the preservation of an occupant's body by stasis. Activates on occupant's unconscious state. Can be locked with medical ID."
+	icon = 'icons/obj/cryobag.dmi'
+	icon_state = "bodybag_folded"
+
+/obj/item/bodybag/cryobag/attack_self(mob/user)
+	var/obj/structure/closet/body_bag/cryobag/R = new /obj/structure/closet/body_bag/cryobag(user.loc)

```

Comment:
Please address this- As it is, this allows for opening cryobags inside places such as spacepods or mecha where they will become completely unrecoverable. Either check to make sure the user's location is a turf, or change this to a get_turf.